redis/src/main/scala/zio/redis/Decoder.scala

```diff
@@ -0,0 +1,7 @@
+package zio.redis
+
+import zio.IO
+
+trait Decoder[A] {
```

Comment:
I am unsure about this. `RespValue` should not leak in the client code. If we're allowing custom decoders (and we should), I think they should work on byte chunks.